functions.php

```diff
@@ -3,18 +3,27 @@
 // On Pi's, this placeholder gets replaced with ${ALLSKY_CONFIG}.
 // On other machines it won't and references to it will silently fail.
 define('ALLSKY_CONFIG',  'XX_ALLSKY_CONFIG_XX');
+// The issue is how to determine if we're on a Pi without using
+// the exec() function which is often disabled on remote machines.
+// And we can't do @exec() to see if it works because that can
+// display a message in the user's browser window.
+// To avoid that message, assume if we're not on a Pi that exec() doesn't work.
+// TODO: make this a user setting if exec() works???
+// If you can think of a better way than to check for a hard-coded
+// path, please update the code.
 
 // If on a Pi, check that the placholder was replaced.
-exec("grep -q 'Model.*: Raspberry' /proc/cpuinfo", $none, $return);
-// Split the placeholder so it doesn't get replaced if the update script is run multiple times.
-// Note: return code 0 == a match, return code 1 == no match
-if ($return==0 && ALLSKY_CONFIG == "XX_ALLSKY_CONFIG" . "_XX") {
+function exec_works() {
+    $disabled = explode(',', ini_get('disable_functions'));
+    return !in_array('exec', $disabled);
+}
+if (exec_works() && ALLSKY_CONFIG == "XX_ALLSKY_CONFIG" . "_XX") {
 	// This file hasn't been updated yet after installation.
 	echo "<div style='font-size: 200%;'>";
 	echo "<span style='color: red'>";
 	echo "Please run the following from the 'allsky' directory before using the Website:";
 	echo "</span>";
-	echo "<code>   website/install.sh --update</code>";
+	echo "<br><br><code>   website/install.sh --update</code>";
 	echo "</div>";
 	exit;
 }
```

Comment:
@EricClaeys I am getting this message even though my website is hosted online. I can setup a testing `php` page in order to test `exec()` and other functions if you want.